functors_task_1/student/include/impl.h

```diff
@@ -40,9 +56,22 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(const AlcoholDrink alcohol,const NonAlcoholDrink nonAlcohol)
     {
+        auto result=Cocktail::Oops;
+        if(AlcoholDrink::Gin==alcohol){
```

Comment:
switch is not good in scope code coverage

Reply:
Nice concern, but could you please provide more info?

Actually, as for me, switching pattern here is needed because it can be easily replaced with a mapping pattern.
I mean a const static map, that represents recipes book. It is very convenient and extendable.

Reply:
pairs code static anaiser and unit test coverage
static - said that default must be included in switch
but if you use default - 100% brunch coverage is hard to reach
so you have to remove one opion or do some other trick... with comment like " analyzer petting"

and yes - swich can works faster and looks better than  if -else if

Reply:
Thanks for the details, but I think it should not be the case for our training.
Readability is the top#1 priority.